subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt_state.c

```diff
@@ -202,13 +248,51 @@ int img_mgmt_get_next_boot_slot(int image, enum img_mgmt_next_boot_type *type)
 	int rcs = img_mgmt_read_info(other_slot, &over, NULL, NULL);
```

Comment:
Just to make sure: this function wouldn't return an error in case there is no proper image on a given slot ?

Reply:
The function check header only, the other stuff is checked by MCUboot.
This function will return error if there is not proper header or read fails.

Reply:
Should we distinguish these 2 error cases?
Unless I missed something, the error related to missing proper header on other slot may prevent code from reaching this check:
```
	if (other_slot_state == DIRECT_XIP_BOOT_UNSET) {
		if (active_slot_state == DIRECT_XIP_BOOT_ONCE) {
			*type = NEXT_BOOT_TYPE_TEST;
		}
		return active_slot;
	}
```

Reply:
I need to take another look at this.